Content.Shared/Chat/TypingIndicator/SharedTypingIndicatorSystem.cs

```diff
@@ -24,15 +18,13 @@ private void OnGotEquipped(EntityUid uid, TypingIndicatorClothingComponent compo
     {
         if (!TryComp<TypingIndicatorComponent>(args.Wearer, out var indicator))
             return;
-
-        indicator.Prototype = component.Prototype;
+        indicator.TypingIndicatorOverrideList.Insert(0, component.OverrideIndicator);
```

Comment:
Instead of shuffling the entire list insert it at the end instead.

Reply:
You both bring up great points but I didn't do either of those for a reason!

I originally wanted to use a queue but the default queue in c# doesn't work for this application without a decent chunk of extra code (When removing a clothing item it needs to remove the first occurrence which isn't necessarily the first thing in the queue). I could fix this by just popping things off until I see the indicator and then putting everything back but that feels like extra code that isn't necessary. I could easily be missing something so let me know if there is a way to make this work!

I also cannot insert the indicator at the end without more extra code:
(A moth equips a lawyer badge and then equips the moth mask and then cardboard robot body.)
[moth, lawyer, moth, robot] (Current indicator is robot)
Remove moth mask:
[lawyer, moth, robot] (Current indicator is robot)
Remove robot:
[lawyer, moth] (Current indicator is moth)
It should be the lawyer and not the default moth indicator.

This could be fixed by removing from the end of the list (Remove starts at the beginning) but that again would require me to write a decent amount of extra code.

I feel like the performance impact of reshuffling the list is really minor and it better to just have a one line solution that is slightly slower than the other two options (Also its not like this is being run every second or anything its kind of uncommon honestly and would probably only happen like ~20 times per round at most).

Let me know what you think!